aggregator/src/messages.rs

```diff
@@ -70,8 +70,10 @@ pub trait TimeExt: Sized {
 }
 
 impl TimeExt for Time {
-    fn as_naive_date_time(&self) -> NaiveDateTime {
-        NaiveDateTime::from_timestamp(self.as_seconds_since_epoch() as i64, 0)
+    fn as_naive_date_time(&self) -> Result<NaiveDateTime, Error> {
+        NaiveDateTime::from_timestamp_opt(self.as_seconds_since_epoch() as i64, 0).ok_or(
```

Comment:
nit: Now that this method is falliable, we could replace the `u64 as i64` conversion with `try_into()`, and propagate an error out.